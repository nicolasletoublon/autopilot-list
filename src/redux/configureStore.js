import thunkMiddleware from 'redux-thunk';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import peopleReducer from './people/reducer';

const middleWares = [thunkMiddleware];

const appReducer = combineReducers({
  people: peopleReducer,
});

const store = createStore(appReducer, applyMiddleware(...middleWares));

export { store };
