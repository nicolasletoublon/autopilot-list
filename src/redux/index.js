/* People REDUCER + ACTIONS */
export { default as list } from './people/reducer';
export { fetchPeopleListThunk } from './people/thunksDispatcher';