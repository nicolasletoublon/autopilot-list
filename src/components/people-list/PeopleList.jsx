import React from 'react';
import styled from 'styled-components';
import PeopleListItem from './PeopleListItem';
import { useDispatch, useSelector } from 'react-redux';
import { fetchPeopleListThunk } from '../../redux';
import ListObserver from './ListObserver';

const ListWrapper = styled.div`
  box-shadow: 0 0 10px 1px #00000054;
  width: 300px;
  border-radius: 5px;
`;

ListWrapper.displayName = 'ListWrapper';

const PeopleListWrapper = styled.ul`
  margin: 0;
  padding: 10px 0;
  list-style-type: none;
`;

const PeopleList = () => {
  const dispatch = useDispatch();
  const { hasMore, peopleList } = useSelector((state) => state.people);

  const fetchMoreData = (inView) => {
    if (inView && hasMore) {
      dispatch(fetchPeopleListThunk());
    }
  };

  return (
    <ListWrapper>
      <PeopleListWrapper>
        {peopleList.map(({ id, ...rest }) => (
          <PeopleListItem key={id} {...rest} />
        ))}
      </PeopleListWrapper>
      {hasMore && <ListObserver onObserverVisible={fetchMoreData} />}
    </ListWrapper>
  );
};

export default PeopleList;
