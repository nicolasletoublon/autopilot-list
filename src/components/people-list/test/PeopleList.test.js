import React from 'react';
import { mount } from 'enzyme';
import PeopleList from '../PeopleList';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

jest.mock('../ListObserver', () => () => 'ListObserver');

describe('PeopleList', () => {
  describe('should render correctly 2 PeopleListItem and no More results', () => {
    const state = {
      people: {
        peopleList: [
          {
            id: '1',
            name: 'name11',
          },
          {
            id: '2',
            name: 'name12',
          },
        ],
      },
    };

    const mockStore = createStore(() => state);

    const comp = (
      <Provider store={mockStore}>
        <PeopleList />
      </Provider>
    );

    it('snapshot', () => {
      const component = mount(comp);
      expect(component).toMatchSnapshot();
    });

    it('should find the 2 list items in the DOM', () => {
      const component = mount(comp);
      expect(component.find('PeopleListItem')).toHaveLength(2);
    });
  });
});
