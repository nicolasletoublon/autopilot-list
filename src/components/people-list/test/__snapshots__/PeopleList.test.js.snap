// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleList should render correctly 2 PeopleListItem and no More results snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PeopleList>
    <ListWrapper>
      <div
        className="sc-AykKH irqydb"
      >
        <styled.ul>
          <ul
            className="sc-AykKI eIKKZe"
          >
            <PeopleListItem
              key="1"
              name="name11"
            >
              <styled.li>
                <li
                  className="sc-AykKE zjrhr"
                >
                  <styled.div>
                    <div
                      className="sc-AykKF eijuMa"
                    >
                      <Adornment
                        name="name11"
                      >
                        <CrossIcon
                          color="#343434"
                          size={15}
                        >
                          <styled.svg
                            className="svg-icon"
                            color="#343434"
                            size={15}
                          >
                            <svg
                              className="sc-AykKC kgFBDO svg-icon"
                              color="#343434"
                              fill="#343434"
                              height="15px"
                              size={15}
                              viewBox="0 0 64 64"
                              width="15px"
                            >
                              <path
                                d="M36.440013,32.0000358 L63.0803642,58.640387 C63.8735026,59.4335253 64.1832584,60.5895496 63.8929496,61.6729968 C63.6026408,62.7564439 62.7563723,63.6027124 61.6729252,63.8930212 C60.589478,64.18333 59.4334537,63.8735742 58.6403154,63.0804358 L31.9999642,36.4400846 L5.35961303,63.0804358 C4.13352741,64.3065214 2.14564984,64.3065214 0.919564218,63.0804358 C-0.306521406,61.8543502 -0.306521406,59.8664726 0.919564218,58.640387 L27.5599154,32.0000358 L0.919564218,5.35968464 C-0.306521406,4.13359901 -0.306521406,2.14572145 0.919564218,0.919635821 C2.14564984,-0.306449804 4.13352741,-0.306449804 5.35961303,0.919635821 L31.9999642,27.559987 L58.6403154,0.919635821 C59.4334537,0.126497436 60.589478,-0.183258385 61.6729252,0.107050386 C62.7563723,0.397359156 63.6026408,1.24362771 63.8929496,2.32707485 C64.1832584,3.41052199 63.8735026,4.56654632 63.0803642,5.35968464 L36.440013,32.0000358 Z"
                              />
                            </svg>
                          </styled.svg>
                        </CrossIcon>
                      </Adornment>
                    </div>
                  </styled.div>
                  <NameWrapper>
                    <div
                      className="sc-AykKG jZCPjH"
                    >
                      name11
                    </div>
                  </NameWrapper>
                </li>
              </styled.li>
            </PeopleListItem>
            <PeopleListItem
              key="2"
              name="name12"
            >
              <styled.li>
                <li
                  className="sc-AykKE zjrhr"
                >
                  <styled.div>
                    <div
                      className="sc-AykKF eijuMa"
                    >
                      <Adornment
                        name="name12"
                      >
                        <CrossIcon
                          color="#343434"
                          size={15}
                        >
                          <styled.svg
                            className="svg-icon"
                            color="#343434"
                            size={15}
                          >
                            <svg
                              className="sc-AykKC kgFBDO svg-icon"
                              color="#343434"
                              fill="#343434"
                              height="15px"
                              size={15}
                              viewBox="0 0 64 64"
                              width="15px"
                            >
                              <path
                                d="M36.440013,32.0000358 L63.0803642,58.640387 C63.8735026,59.4335253 64.1832584,60.5895496 63.8929496,61.6729968 C63.6026408,62.7564439 62.7563723,63.6027124 61.6729252,63.8930212 C60.589478,64.18333 59.4334537,63.8735742 58.6403154,63.0804358 L31.9999642,36.4400846 L5.35961303,63.0804358 C4.13352741,64.3065214 2.14564984,64.3065214 0.919564218,63.0804358 C-0.306521406,61.8543502 -0.306521406,59.8664726 0.919564218,58.640387 L27.5599154,32.0000358 L0.919564218,5.35968464 C-0.306521406,4.13359901 -0.306521406,2.14572145 0.919564218,0.919635821 C2.14564984,-0.306449804 4.13352741,-0.306449804 5.35961303,0.919635821 L31.9999642,27.559987 L58.6403154,0.919635821 C59.4334537,0.126497436 60.589478,-0.183258385 61.6729252,0.107050386 C62.7563723,0.397359156 63.6026408,1.24362771 63.8929496,2.32707485 C64.1832584,3.41052199 63.8735026,4.56654632 63.0803642,5.35968464 L36.440013,32.0000358 Z"
                              />
                            </svg>
                          </styled.svg>
                        </CrossIcon>
                      </Adornment>
                    </div>
                  </styled.div>
                  <NameWrapper>
                    <div
                      className="sc-AykKG jZCPjH"
                    >
                      name12
                    </div>
                  </NameWrapper>
                </li>
              </styled.li>
            </PeopleListItem>
          </ul>
        </styled.ul>
      </div>
    </ListWrapper>
  </PeopleList>
</Provider>
`;
